bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/DialogProcessor.java

```diff
@@ -81,28 +88,42 @@ public DialogProcessor(KSService ks, STTService stt, TTSService tts, HumanLangua
         this.source = source;
         this.sink = sink;
         this.keyword = keyword;
+        this.listeningItem = listeningItem;
+        this.eventPublisher = eventPublisher;
         this.format = AudioFormat.getBestMatch(source.getSupportedFormats(), sink.getSupportedFormats());
     }
 
     public void start() {
         try {
             ks.spot(this, source.getInputStream(format), locale, this.keyword);
         } catch (KSException | AudioException e) {
-            logger.error("Encountered error calling spot: {}", e.getMessage());
+            logger.error("Encountered error calling spot: {}", e);
+        }
+    }
+
+    private void toggleProcessing(boolean value) {
+        if (this.processing == value) {
+            return;
+        }
+        this.processing = value;
+        if (listeningItem != null && ItemUtil.isValidItemName(listeningItem)) {
+            OnOffType command = (value) ? OnOffType.ON : OnOffType.OFF;
+            eventPublisher.post(ItemEventFactory.createCommandEvent(listeningItem, command));
         }
     }
 
     @Override
     public void ksEventReceived(KSEvent ksEvent) {
         if (!processing) {
-            processing = true;
             this.isSTTServerAborting = false;
             if (ksEvent instanceof KSpottedEvent) {
+                toggleProcessing(true);
                 if (stt != null) {
                     try {
                         this.sttServiceHandle = stt.recognize(this, source.getInputStream(format), this.locale,
                                 new HashSet<String>());
                     } catch (STTException | AudioException e) {
+                        logger.error("Error during recognition", e);
```

Comment:
Actually I had initially an exception coming from the `source.getInputStream(format)` call (wrong capture frequency IIRC). So I only had the message "Error creating the audio input stream" without the cause. Maybe it makes sense to only log an `AudioException`?